### PR TITLE
Changed monitoring metrics structure

### PR DESCRIPTION
**Pull request by Kanikakhetawat**

### CODE REVIEW

File: src/main/java/io/appform/dropwizard/sharding/metrics/EntityOpMetricKey.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package io.appform.dropwizard.sharding.metrics;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+
+@Value
+@Builder
+@EqualsAndHashCode
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/metrics/EntityOpMetricKey.java:9]: Remove this

File: src/main/java/io/appform/dropwizard/sharding/metrics/TransactionMetricObserver.java [hunk 2]
```diff
@@ -57,23 +50,15 @@ public <T> T execute(TransactionExecutionContext context, Supplier<T> supplier)
...
-        val daoClass = context.getDaoClass();
-        val daoMetricPrefix = daoMetricPrefixCache.computeIfAbsent(daoClass, key -> {
-            val prefix = metricManager.getDaoMetricPrefix(daoClass);
-            daoToOpTypeMetricCache.put(prefix, new HashMap<>());
-            return prefix;
-        });
-        val opType = context.getOpType();
-        return daoToOpTypeMetricCache.get(daoMetricPrefix).computeIfAbsent(opType,
-                key -> metricManager.getDaoOpMetricData(daoMetricPrefix, context));
+        return Lists.newArrayList(entityOpMetricData, shardMetricData);
```
Comment (santanusinha) [inline src/main/java/io/appform/dropwizard/sharding/metrics/TransactionMetricObserver.java:62]: Create immutable list

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: src/main/java/io/appform/dropwizard/sharding/metrics/EntityOpMetricKey.java (hunk 1 above)
Comment (Kanikakhetawat) [reply src/main/java/io/appform/dropwizard/sharding/metrics/EntityOpMetricKey.java]: Removed

File: src/main/java/io/appform/dropwizard/sharding/metrics/TransactionMetricObserver.java (hunk 2 above)
Comment (Kanikakhetawat) [reply src/main/java/io/appform/dropwizard/sharding/metrics/TransactionMetricObserver.java]: Done

**Review by santanusinha — APPROVED** [review]

**Pull request merged**